### PR TITLE
Build command first refactor

### PR DESCRIPTION
**Pull request by ifbyol**
This is a first PR on the work to refactor the build part of the CLI. In this PR there are a few changes:
* Removed the builder that was under the `cmd/build/remote`. This was only used for the deploy and destroy remote but the logic in the builder wasn't really be needed for the remote scenario as we are the ones setting all the build options accordingly. In this way, we also stop tracking remote deploy and destroy as builds, as the logic for tracking was in this builder.
* Deploy and destroy remote have a dependency now with a remoteRunner instead of a builder. The idea is to make agnostic the commands about where the commands really run. It is not completely agnostic because the function params use `BuildOptions`, but it is a first step.
* All the logic related to build that was on v1 builder is now extracted to a "basic" builder (we can look for a better name). The idea is that the basic builder has the minimum logic to build an image and then, can be used by more complex builders to extend the functionality
* v1 builder is a wrapper of the basic builder which basically runs the build but also have the `IsV1` function (used in other parts of the code).
* v2 builder is also a wrapper of the basic builder adding all the logic about smart builds and git calculation. In this way, we break the dependency between v1 and v2 builders. They both use a basic builder
* Also, this PR fixes a bug in the remote destroy when the command is executed from a parent folder of the okteto manifest. The deploy was fixed in this [PR](https://github.com/okteto/okteto/pull/4128) but we didn't include that logic in the destroy
* Removed some unused dependencies in the v1 builder (`OktetoBuilderInterface` and `oktetoRegistryInterface`)

I created a couple of simple diagrams to understand the changes. Before the changes in this PR, the dependency/relationship between builders was the following:

Screenshot 2024-02-29 at 14 54 18

After the changes in this PR, the relationships are:

Screenshot 2024-02-29 at 14 54 29

### CODE REVIEW

File: cmd/build/v2/build_test.go [hunk 1]
```diff
@@ -158,15 +159,14 @@ func (a *fakeAnalyticsTracker) TrackImageBuild(meta ...*analytics.ImageBuildMeta
 	a.metaPayload = meta
 }
 
-func NewFakeBuilder(builder OktetoBuilderInterface, registry oktetoRegistryInterface, cfg oktetoBuilderConfigInterface, analyticsTracker analyticsTrackerInterface) *OktetoBuilder {
+func NewFakeBuilder(builder buildCmd.OktetoBuilderInterface, registry oktetoRegistryInterface, cfg oktetoBuilderConfigInterface, analyticsTracker analyticsTrackerInterface) *OktetoBuilder {
 	return &OktetoBuilder{
-		Registry:          registry,
-		Builder:           builder,
+		Registry: registry,
+		// Builder:           builder,
```
Comment (andreafalzetti) [inline cmd/build/v2/build_test.go:165]:
```suggestion
```

Should we remvoe this?

**Review by andreafalzetti — APPROVED** [review]
Great change 👏

File: go.mod [hunk 2]
```diff
@@ -237,6 +237,7 @@ require (
 	github.com/samber/lo v1.38.1 // indirect
 	github.com/samber/slog-common v0.11.0 // indirect
 	github.com/skeema/knownhosts v1.2.1 // indirect
+	github.com/stretchr/objx v0.5.0 // indirect
```
Comment (jLopezbarb) [inline go.mod:240]: Why do we need this change? I don't see any use on the code and we didn't update any dependency

File: cmd/destroy/remote.go [hunk 3]
```diff
@@ -117,19 +123,24 @@ type remoteDestroyCommand struct {
 
 func newRemoteDestroyer(manifest *model.Manifest, ioCtrl *io.Controller) *remoteDestroyCommand {
 	fs := afero.NewOsFs()
-	builder := remoteBuild.NewBuilderFromScratch(ioCtrl)
+	runner := &buildCmd.OktetoBuilder{
```
Comment (jLopezbarb) [inline cmd/destroy/remote.go:126]: NIT: Could we move this into a init function? I think right now is not a problem but if we ever end up adding more fields to buildCmd.OktetoBuilder, we can forget to add that field on all the initializations. WDYT?

File: go.mod (hunk 2 above)
Comment (ifbyol) [reply go.mod:240]: I'm not sure, I just run go mod tidy from the vscode and it was added. I will delete it and executed again to see if it pops up again

File: cmd/build/v2/build_test.go (hunk 1 above)
Comment (ifbyol) [reply cmd/build/v2/build_test.go:165]: Yes, good catch!

File: cmd/destroy/remote.go (hunk 3 above)
Comment (ifbyol) [reply cmd/destroy/remote.go]: I created the init function but that doesn't prevent you from that error. You can always create the OktetoBuilder outside the init function for whatever reason

File: cmd/build/v2/build_test.go (hunk 1 above)
Comment (ifbyol) [reply cmd/build/v2/build_test.go]: Removed: https://github.com/okteto/okteto/pull/4189/commits/ffb513fbeb752fe766c03aea2cd3e0090764ac94

**Review by andreafalzetti — APPROVED** [review]

**Review by jLopezbarb — APPROVED** [review]

**Pull request merged**